### ISSUE TITLE
 Claude-3-Opus (You) error: Failed to connect to browser

### ISSUE THREAD

**Issue by rqi14** [xtekky/gpt4free#1929]
**Bug description**
I asked a question using claude-3-opus model from You. It reports error:

`An error occured: You: Exception: --------------------- Failed to connect to browser --------------------- One of the causes could be when you are running as root. In that case you need to pass no_sandbox=True`

**Environment**
- 0.3.10 (docker compose)

**Comment by hlohaus** [comment]
The browser won't open. Maybe it's out of memory or you're doing too many things at once.

**Comment by rqi14** [comment]
> The browser won't open. Maybe it's out of memory or you're doing too many things at once.

Hi. I have 128GB of memory and at least 100GB are free. It doesn't look like a memory issue. I run it using docker compose. Is it somehow related to that?

**Comment by rqi14** [comment]
> The browser won't open. Maybe it's out of memory or you're doing too many things at once.

By checking the noVNC for the VM, I found the browser was actually open but showed a blank page.

**Comment by Looksh190** [comment]
Only gpt4 for now:(

**Comment by hlohaus** [comment]
Please confirm if the following line is present in your docker-compose.yml file:
    shm_size: 2gb

It is important to note that Google Chrome requires a minimum amount of memory to function properly.

Alternatively, you can try restarting the container and running nodriver again.

**Comment by rqi14** [comment]
> Please confirm if the following line is present in your docker-compose.yml file: shm_size: 2gb
> 
> It is important to note that Google Chrome requires a minimum amount of memory to function properly.
> 
> Alternatively, you can try restarting the container and running nodriver again.

Hi.

I double checked and can confirm the 'shm_size: 2gb' option is present in the docker-compose file. I don't quite understand the nodriver thing. It does not seem to be present in any of the document. 

Please see the below screenshot. I think the chrome is open but it shows a blank page.

![image](https://github.com/xtekky/gpt4free/assets/26152437/2eec21d5-4505-478c-a181-c9033cc8ecb7)

**Comment by KGLongWang** [comment]
I have the same question  no_sandbox=True

**Comment by daniilshamraev** [comment]
I have the same question no_sandbox=True (Need for Heroku hosting)

**Comment by Antom91** [comment]
I have the same question no_sandbox=True

**Comment by rqi14** [comment]
> Please confirm if the following line is present in your docker-compose.yml file: shm_size: 2gb
> 
> It is important to note that Google Chrome requires a minimum amount of memory to function properly.
> 
> Alternatively, you can try restarting the container and running nodriver again.

Hi. I probably found the cause of the issue. I cannot access you.com directly. It must happen via a proxy. Therefore, in addition to the G4F_PROXY environment varaible, I set HTTP_PROXY and HTTPS_PROXY. If these two are not set, the chromium window pops up with the url being you.com but it is inaccessible due to my network enviroment. If I set these two proxy options, I have the problem that the chromium window pops up with a empty blank page.
